src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayFileSystem.cs

```diff
@@ -59,36 +64,74 @@ public void WriteMergedOCIFileLayer(OCIFileLayer oneLayer)
 
         public List<OCIArtifactLayer> ReadImageLayers()
         {
-            return ReadOCIArtifactLayers(WorkingImageLayerFolder);
+            var manifest = ReadManifest();
+            var imageLayers = ReadOCIArtifactLayers(WorkingImageLayerFolder);
+            var sortedLayers = new List<OCIArtifactLayer>();
+            ValidationUtility.ValidateManifest(manifest);
+            foreach (var layerInfo in manifest.Layers)
+            {
+                var currentLayers = imageLayers.Where(layer => layer.Digest == layerInfo.Digest).ToList();
+                if (!currentLayers.Any())
+                {
+                    throw new OverlayException(TemplateManagementErrorCode.ImageLayersNotFound, $"Layer {layerInfo.Digest} not found.");
+                }
+
+                sortedLayers.Add(currentLayers[0]);
+            }
+
+            return sortedLayers;
         }
 
         public void WriteImageLayers(List<OCIArtifactLayer> imageLayers)
         {
             EnsureArg.IsNotNull(imageLayers, nameof(imageLayers));
 
             ClearFolder(WorkingImageLayerFolder);
+
+            // Used to label layers sequence. In this version, only two layers will be generated.
+            // Should not exceed 9 in the future.
+            var layerNumber = 1;
             foreach (var layer in imageLayers)
             {
-                layer.WriteToFolder(WorkingImageLayerFolder);
+                if (layer.Content == null)
+                {
+                    continue;
+                }
+
+                layer.WriteToFile(Path.Combine(WorkingImageLayerFolder, string.Format("layer{0}.tar.gz", layerNumber)));
+                layerNumber += 1;
             }
         }
 
-        public List<OCIArtifactLayer> ReadBaseLayers()
+        public OCIArtifactLayer ReadBaseLayer()
```

Comment:
i feel we can move the function to operator if it doesn't have relationship with filesystem